docs/en/observability/logs-stream.asciidoc

```diff
@@ -855,4 +870,141 @@ PUT /logs-example-default/_mapping
     }
   }
 }
+----
+
+[discrete]
+[[logs-stream-reroute]]
+= Reroute log data to specific data stream
+
+preview::[]
+
+By default, an ingest pipeline sends your log data to a single data stream. To simplify log data management, you can use a {ref}/reroute-processor.html[reroute processor] to route data from the generic data stream to a target data stream. For example, you might want to send high-severity logs to a specific data stream that's different from low-severity logs to help with categorization. 
+
+This section shows you how to use a reroute processor to send the high-severity logs (`WARN` or `ERROR`) from the following examples to a specific data stream and keep regular logs (`DEBUG` and `INFO`) in the default data stream:
+
+[source,log]
+----
+2023-08-08T13:45:12.123Z WARN 192.168.1.101 Disk usage exceeds 90%.
+2023-08-08T13:45:14.003Z ERROR 192.168.1.103 Database connection failed.
+2023-08-08T13:45:15.004Z DEBUG 192.168.1.104 Debugging connection issue.
+2023-08-08T13:45:16.005Z INFO 192.168.1.102 User changed profile picture.
+----
+
+To use a reroute processor:
+
+. <<logs-stream-reroute-pipeline, Add a reroute processor to your ingest pipeline.>>
+. <<logs-stream-reroute-add-logs, Add the example logs to your data stream.>>
+. <<logs-stream-reroute-verify, Query your logs and verify the high-severity logs were routed to the new data stream.>>
+
+[discrete]
+[[logs-stream-reroute-pipeline]]
+=== Add a reroute processor to your ingest pipeline
+
+You can add a reroute processor to your ingest pipeline with the following command:
+
+[source,console]
+----
+PUT _ingest/pipeline/logs-example-default
+{
+  "description": "Extracts fields and reroutes WARN",
+  "processors": [
+    {
+      "dissect": {
+        "field": "message",
+        "pattern": "%{@timestamp} %{log.level} %{host.ip} %{message}"
+      },
+      "reroute": {
+        "tag": "high_severity_logs",
+        "if" : "ctx.log?.level == 'WARN' || ctx.log?.level == 'ERROR'",
```

Comment:
@felixbarny Even though the example is correct not sure  if we should encourage to route on severity levels for logs? I rather use an example with service.name?

Reply:
I don't think that's a bad example to be honest. One thing I would add is to add a recommendation to keep the number of data streams relatively low. We can link to this page for more details: https://www.elastic.co/guide/en/elasticsearch/reference/current/size-your-shards.html.

Also, we should probably advocate for using the field API here

```suggestion
        "if" : "$('log.level', '') == 'WARN' || $('log.level', '') == 'ERROR'",
```

Reply:
@mdbirnstiehl have you tested that this syntax works? I've just added that suggestion from how I remember the syntax.

I'd also like to see a follow up PR that links to https://www.elastic.co/guide/en/elasticsearch/reference/current/size-your-shards.html.

Reply:
I'll verify the syntax and update accordingly. I added a note on line 893 linking to the size your shards doc